common/v2/components/Dropdown.tsx

```diff
@@ -29,48 +36,88 @@ interface Props<T> {
   disabled?: boolean;
   placeholder?: string;
   searchable?: boolean;
+  clearable?: boolean;
   name?: string;
+  dropdownIcon?: JSX.Element;
   optionComponent?:
     | React.ComponentClass<OptionComponentProps<T>>
     | React.StatelessComponent<OptionComponentProps<T>>;
   valueComponent?: React.ComponentClass<T> | React.StatelessComponent<T>;
+  inputValue?: string;
+  onInputChange?: OnInputChangeHandler;
+  onInputKeyDown?: OnInputKeyDownHandler;
   onChange?(option: T): void;
+  onBlur?(e: string | undefined): void;
 }
 
 const Chevron = styled(Icon)`
   font-size: 0.75rem;
 `;
+const IconWrapper = styled('div')`
+  width: 30px;
+`;
 
 const DropdownIndicator = (props: ArrowRendererProps) => (
   <Chevron icon={props.isOpen ? 'chevronUp' : 'chevronDown'} />
 );
 
+const CustomDropdownIndicator = (dropdownIcon: JSX.Element) => () => (
+  <IconWrapper>{dropdownIcon}</IconWrapper>
+);
+
+const ClearIndicator = () => (
+  <IconWrapper>
+    <img src={crossIcon} />
+  </IconWrapper>
+);
+
+// AccountLookup dropdown is using custom dropdown indicator (dropdownIcon) and clear field indicator.
+// When value is set, it hides dropdown icon, so that clear icon can appear instead of it.
+const getDropdownIndicator = (
+  value: Props<any>['value'],
+  dropdownIcon: Props<any>['dropdownIcon']
+) => {
+  if (!dropdownIcon) return DropdownIndicator;
+  if (!value) return CustomDropdownIndicator(dropdownIcon);
+  return null;
+};
+
 export default function Dropdown({
   onChange,
+  onBlur,
   options,
   optionComponent,
   value,
   valueComponent,
   placeholder,
   disabled,
   searchable,
-  name // field name for hidden input. Important for Formik
+  clearable = false,
+  inputValue,
+  onInputChange,
+  name, // field name for hidden input. Important for Formik
+  onInputKeyDown,
+  dropdownIcon
```

Comment:
Maybe it's time we just pass through the props? `{...props}`

Reply:
Agree